api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -369,14 +369,24 @@ const (
 	PruneRetentionPolicy RetentionPolicy = "Prune"
 )
 
-// +kubebuilder:validation:Enum=ImageRegistry
-type OptionalCapability string
-
-const ImageRegistryCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityImageRegistry)
-
-// capabilities allows disabling optional components at install time.
+// capabilities allows enabing or disabling optional components at install time.
 // Once set, it cannot be changed.
+// self.set1.all(e, !(e in self.set2)),
+//
+// +kubebuilder:validation:XValidation:rule="self.enabled.all(e, !(e in self.disabled))", message="Capabilities can not be both enabled and disabled at once."
```

Comment:
CRD is invalid...error

Reply:
I'm not overly experienced in advanced rules. In fact, for a rule so complex, I would go for the validation webhook (if the Hypershift API has any). I see that they do use quite complex rules here: https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/nodepool_types.go#L86-L89

The `enabled` and `disabled` fields are optional, I wonder if you need to use `has` to check for their presence. Your failure happens at the path `spec.validation.openAPIV3Schema.properties[spec].properties[capabilities].default`, and the `default` bit makes me suspect that an empty object no longer passes your validation. So what about

```
has(self.enabled) && has(self.disabled) && self.enabled.all(e, !(e in self.disabled))
```

?